crates/supervisor/storage/src/log/mod.rs

```diff
@@ -0,0 +1,65 @@
+//! Log storage abstraction for supervisor state tracking.
+//!
+//! This module defines traits for reading and writing logs and block metadata
+//! associated with L2 execution. These traits are implemented by concrete
+//! backends like [`mdbx`] and are used by the supervisor during block ingestion
+//! and verification workflows.
+
+mod mdbx;
+
+use kona_protocol::BlockInfo;
+use kona_supervisor_types::Log;
+
+/// Provides read-only access to log and block metadata stored in the supervisor database.
+///
+/// This trait is used by components that need to inspect the state of execution logs
+/// and block metadata without modifying storage.
+#[allow(dead_code)]
```

Comment:
```suggestion
#[expect(dead_code)]
```
`expect` will emit a warning when the lint expectation is no longer fulfilled, this helps us maintain the code so we don't leave redundant `allow` attributes in the codebase